lib/local_authorities_url_importer.rb

```diff
@@ -0,0 +1,93 @@
+require 'csv'
+
+class LocalAuthoritiesUrlImporter
+  CONTACTS_LIST_URL = "http://local.direct.gov.uk/Data/local_authority_contact_details.csv"
+
+
+  def self.import_urls
+    fh = fetch_data
+    begin
+      new(fh).import
+    ensure
+      fh.close
+    end
+  end
```

Comment:
We can rejig `fetch_http_to_file` to take a block and avoid having to write this `begin...ensure...end`.  If in `fetch_http_to_file` we used [`Tempfile.open`](http://ruby-doc.org/stdlib-2.3.0/libdoc/tempfile/rdoc/Tempfile.html#method-c-open) instead of `.new` we can pass it a block and it will deal with closing and unlinking the file automatically.  Something like:

```ruby
def self.import_urls
  fetch_data do |fh|
    new(fh).import
  end
end

def self.fetch_data(&block)
  block.call fetch_http_to_file(CONTACTS_LIST_URL)
end

def self.fetch_http_to_file(url, &block)
  Tempfile.open(%w(local_authority_data csv) do |fh|
    # existing code
    block.call(fh)
  end
end
```

It doesn't really save us much typing, but it is a wee bit more idiomatic for rubyists to use blocks (especially when dealing with files) to allow the ruby stdlib to deal with cleanup for us.